core/lib/web3_decl/src/client/shared.rs

```diff
@@ -0,0 +1,89 @@
+//! `Shared` RPC client.
+
+use std::{fmt, sync::Arc};
+
+use async_trait::async_trait;
+use jsonrpsee::core::{
+    client::{BatchResponse, ClientT, Error, Subscription, SubscriptionClientT},
+    params::BatchRequestBuilder,
+    traits::ToRpcParams,
+};
+use serde::de::DeserializeOwned;
+
+#[derive(Debug)]
+pub struct Shared<T>(Arc<T>);
```

Comment:
Do we need it because of orphan rule?

Reply:
Yep, both `Arc` and the traits we want to implement are defined externally.